opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -231,6 +231,9 @@ def __init__(
                     "Invalid key/value pair (%s, %s) found.", key, value
                 )
 
+    def __contains__(self, item: object) -> bool:
+        return item in self._dict
+
     def __getitem__(self, key: str) -> typing.Optional[str]:  # type: ignore
         return self._dict.get(key)
 
```

Comment:
Or we could change the `__getitem__` to 
```python
    def __getitem__(self, key: str) ->str:
         return self._dict[key]
```
It raises KeyError similar to other mapping types when there is no key and also fixes the issue. The current `__getitem__` returning None when there is no key contributed to incorrect behaviour.

Reply:
I see what you mean now that I'm looking, and I added it to the PR with some more tests.  Opinions on keeping the `__contains__`?

Or would you like to subclass this instead, as @ocelotl suggested?

Reply:
> would you like to subclass this instead, as @ocelotl suggested?

I think there were some issues with typing and OrderedDict for lower python versions(< py3.7.x). If this is not a problem anymore we are better off subclassing than to implement the dunder methods.

> Opinions on keeping the __contains__?

No strong opinions, If there is consensus with getitem change I don't find contains adding much value. But I am also fine with keeping it.